src/utils.ts

```diff
@@ -0,0 +1,27 @@
+// Genel fetchData fonksiyonu sadece URL kabul eder
+export async function fetchData<T>(url: string): Promise<T> {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`An error occurred when fetching data from ${url}.`);
+  }
+  return await response.json();
+}
+
+// Wrapper fonksiyonu: URL ve idMeal'i birleştirir ve fetchData'yı çağırır
+export const fetchRecipeById = <T>(url: string, idMeal: string): Promise<T> => {
```

Comment:
Since this function is too specific to recipes, it should be moved to recipeService or api.ts file (whatever you decide to keep after merging them into one).